kernel/kernel-api/src/main/java/io/delta/kernel/internal/checkpoints/CheckpointInstance.java

```diff
@@ -30,13 +31,25 @@ public class CheckpointInstance
     /**
      * Placeholder to identify the version that is always the latest on timeline
      */
-    public static final CheckpointInstance MAX_VALUE = new CheckpointInstance(-1);
+    public static final CheckpointInstance MAX_VALUE = new CheckpointInstance(Long.MAX_VALUE);
 
     public final long version;
     public final Optional<Integer> numParts;
 
     public CheckpointInstance(Path path) {
-        this(FileNames.getFileVersion(path));
+        String[] pathParts = path.getName().split("\\.");
```

Comment:
Is there way to work without using the `Path`. Currently, the `Path` class is a copy of the Hadoop `Path` and it is just a lot of code which we don't need.

Reply:
I think no reason to use Path here. We convert from String to Path in `SnapshotManager`. Will update